pdns/dnsparser.cc

```diff
@@ -26,74 +26,56 @@
 
 #include "namespaces.hh"
 
-class UnknownRecordContent : public DNSRecordContent
+UnknownRecordContent::UnknownRecordContent(const string& zone) 
 {
-public:
-  UnknownRecordContent(const DNSRecord& dr, PacketReader& pr) 
-    : d_dr(dr)
-  {
-    pr.copyRecord(d_record, dr.d_clen);
+  // parse the input
+  vector<string> parts;
+  stringtok(parts, zone);
+  // we need exactly 3 parts, except if the length field is set to 0 then we only need 2
+  if (parts.size() != 3 && !(parts.size() == 2 && equals(parts.at(1), "0"))) {
+    throw MOADNSException("Unknown record was stored incorrectly, need 3 fields, got " + std::to_string(parts.size()) + ": " + zone);
   }
 
-  UnknownRecordContent(const string& zone) 
-  {
-    // parse the input
-    vector<string> parts;
-    stringtok(parts, zone);
-    // we need exactly 3 parts, except if the length field is set to 0 then we only need 2
-    if (parts.size() != 3 && !(parts.size() == 2 && equals(parts.at(1), "0"))) {
-      throw MOADNSException("Unknown record was stored incorrectly, need 3 fields, got " + std::to_string(parts.size()) + ": " + zone);
-    }
-
-    if (parts.at(0) != "\\#") {
-      throw MOADNSException("Unknown record was stored incorrectly, first part should be '\\#', got '" + parts.at(0) + "'");
-    }
+  if (parts.at(0) != "\\#") {
+    throw MOADNSException("Unknown record was stored incorrectly, first part should be '\\#', got '" + parts.at(0) + "'");
+  }
 
-    const string& relevant = (parts.size() > 2) ? parts.at(2) : "";
-    unsigned int total = pdns_stou(parts.at(1));
-    if (relevant.size() % 2 || (relevant.size() / 2) != total) {
-      throw MOADNSException((boost::format("invalid unknown record length: size not equal to length field (%d != 2 * %d)") % relevant.size() % total).str());
-    }
+  const string& relevant = (parts.size() > 2) ? parts.at(2) : "";
+  unsigned int total = pdns_stou(parts.at(1));
+  if (relevant.size() % 2 || (relevant.size() / 2) != total) {
+    throw MOADNSException((boost::format("invalid unknown record length: size not equal to length field (%d != 2 * %d)") % relevant.size() % total).str());
+  }
 
-    string out;
-    out.reserve(total + 1);
+  string out;
+  out.reserve(total + 1);
 
-    for (unsigned int n = 0; n < total; ++n) {
-      int c;
-      if (sscanf(&relevant.at(2*n), "%02x", &c) != 1) {
-        throw MOADNSException("unable to read data at position " + std::to_string(2 * n) + " from unknown record of size " + std::to_string(relevant.size()));
-      }
-      out.append(1, (char)c);
+  for (unsigned int n = 0; n < total; ++n) {
```

Comment:
you could use boost::algorithm::unhex here..

```.c
std::stringstream ss;
boost::algorithm::unhex(relevant, std::ostream_iterator<char>(ss));
out = ss.str();
```

Reply:
none of this is new code, and i'm not going to change it.